test/e2e/tests/hardware-wallets/ledger-erc20.spec.ts

```diff
@@ -43,6 +49,62 @@ describe('Ledger Hardware', function (this: Suite) {
         await testDappPage.check_TokenAddressesValue(
           '0xcB17707e0623251182A654BEdaE16429C78A7424',
         );
+        // Add to wallet
+        await testDappPage.clickERC20WatchAssetButton();
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        const watchAssetConfirmation = new WatchAssetConfirmation(driver);
+        await watchAssetConfirmation.clickFooterConfirmButton();
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+        const homePage = new HomePage(driver);
+        await homePage.goToTokensTab();
+        await homePage.check_expectedTokenBalanceIsDisplayed('10', symbol);
+
+        // Transfer token
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await testDappPage.clickERC20TokenTransferButton();
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        const tokenTransferTransactionConfirmation =
+          new TokenTransferTransactionConfirmation(driver);
+        await tokenTransferTransactionConfirmation.check_pageIsLoaded();
+        await tokenTransferTransactionConfirmation.clickConfirmButton();
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+        const activityListPage = new ActivityListPage(driver);
+        await homePage.goToActivityList();
+        await activityListPage.check_txAction(`Sent ${symbol}`);
+        await activityListPage.check_txAmountInActivity(`-1.5 ${symbol}`);
+
+        // Approve token
+        const txConfirmation = new TransactionConfirmation(driver);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await testDappPage.clickApproveTokens();
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        await txConfirmation.clickFooterConfirmButton();
```

Comment:
could we use this method instead, to prevent flakiness?

```suggestion
        await txConfirmation.clickFooterConfirmButtonAndAndWaitForWindowToClose();
```